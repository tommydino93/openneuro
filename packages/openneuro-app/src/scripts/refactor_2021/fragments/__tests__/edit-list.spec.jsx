import React from 'react'
import { mount } from 'enzyme'
import EditList from '../edit-list.jsx'

// Mount is used because enzyme does not yet support hooks
describe('EditList component', () => {
  it('renders with default props', () => {
    const wrapper = mount(<EditList />)
    expect(wrapper).toMatchSnapshot()
  })
  it('calls setElements when an item is added', () => {
    const setElements = jest.fn()
    const wrapper = mount(<EditList setElements={setElements} />)
    wrapper.find('button[children="Add"]').simulate('click')
    // Added an empty string to an empty list
    expect(setElements).toHaveBeenCalledWith([''])
  })
  it('renders any elements', () => {
    const wrapper = mount(<EditList elements={['One', 'Two']} />)
    const values = wrapper.find('span.input-group-addon')
    expect(values).toHaveLength(2)
    expect(values.at(0).text()).toBe('One')
    expect(values.at(1).text()).toBe('Two')
  })
})
