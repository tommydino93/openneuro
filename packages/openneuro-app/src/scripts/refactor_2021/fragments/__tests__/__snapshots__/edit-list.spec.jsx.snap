// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditList component renders with default props 1`] = `
<EditList>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
    <span
      className="input-group-btn"
    >
      <Styled(button)
        className="btn btn-default"
        onClick={[Function]}
        type="button"
      >
        <button
          className="btn btn-default css-dhocte"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </Styled(button)>
    </span>
  </div>
  <div />
</EditList>
`;
